advanced_source/dispatcher.rst

```diff
@@ -179,6 +182,17 @@ functions:
   :start-after: BEGIN TORCH_LIBRARY_IMPL Autograd
   :end-before: END TORCH_LIBRARY_IMPL Autograd
 
+Here in the example we register the kernel to ``Autograd``, which is an alias
+dispatch key that propagates the kernel to all Autograd backend keys, e.g. ``AutogradCPU``,
+``AutogradCUDA``, ``AutogradXLA``, ``AutogradOther`` etc.  The propagation happens
+in dispatch table precomputation when the process starts up.  Optionally you
+can register optimized autograd kernel for a backend (e.g. CPU) to its corresponding
+Autograd dispatch key (e.g. AutogradCPU) as well.  When registration to ``AutogradCPU``
+and ``Autograd`` both exist, the direct registration to ``AutogradCPU`` takes precedence.
+To better understand the precompute logic and precedence among all dispatch keys, you can use
+``PythonDispatcher`` provided in `tools/python_dispatcher.py <https://github.com/pytorch/pytorch/blob/master/tools/python_dispacher.py`_.
+For a certain operator, it shows how the computed dispatch table looks like after your registrations.
```

Comment:
There is a sort of pedagogical question here, which is whether or not to pretend Autograd is a normal dispatch key in the initial explanation, and then only later explain alias keys as a refinement to the model, or attempt to talk about alias keys immediately from the getgo. Personally, I think it makes more sense to not mention alias keys until later (or maybe even not at all in this document), because the average user probably only needs cookbook style understanding of dispatch keys (if I want to do X, then I want key Y) and wouldn't benefit all that much from understanding the underlying model (in part because they don't actually have the ability to affect it, e.g., define new alias keys).

Reply:
I'd agree that we don't need to point the full firehose at them here, but I think we miss an opportunity by not mentioning it at all - first because we're already opening the door (rightly) with "corresponding Autograd key" above, and second because if they wind up using PythonDispatcher they're going to be better equipped to understand what's going on if we've primed them at least a little here. 

What about a sidebar that's a bit more abbreviated, something like

```
.. note::
    
    In this example we register the kernel to ``Autograd``, which installs it as the
    autograd kernel for all backends. You can also register optimized kernels for specific
    backends by using the corresponding backend-specific dispatch key - for example, 
    ``AutogradCPU`` or ``AutogradCUDA``. To explore these and other dispatch key 
    options in more detail, check out the ``PythonDispatcher`` tool provided in 
    `tools/python_dispatcher.py <https://github.com/pytorch/pytorch/blob/master/tools/python_dispacher.py`_.

```

Note: there's a typo in the link to python_dispatcher.py but I don't know how to fix it 😬

Reply:
Aha, I was actually thinking the shorter note could replace the original paragraph above (it was a way of addressing Ed's concern of not digging into alias key details too early, but giving enough context that they'll know what's going on when they encounter them later). 

But I'd either stick with the original paragraph or use the new note - they're really two versions of the same content.